src/poetry/inspection/info.py

```diff
@@ -450,7 +457,7 @@ def _get_poetry_package(path: Path) -> ProjectPackage | None:
         return None
 
     @classmethod
-    def _pep517_metadata(cls, path: Path) -> PackageInfo:
+    def _pep517_metadata(cls, path: Path) -> PackageInfo | None:
```

Comment:
Is raising exception the same as returning `None`?

Reply:
Looks like you already addressed this as part of #5601 💯